stacks/bracketstester.js

```diff
@@ -0,0 +1,30 @@
+const Stack = require('./Stack');
+
+function BracketsTester(stackMaxCapacity) {
+    this.stringStack = new Stack(stackMaxCapacity);
+    this.count = 0;
+}
+
+BracketsTester.prototype.checkBrackets = function(stringValue) {
+
+    Array.from(stringValue).forEach(charValue => this.stringStack.push(charValue));
+
+    while ( this.stringStack.peek() ) {
```

Comment:
you are using falsy as check a check for going forward but what else can be falsy that you can get in your function
1. 0 or "0"
2. " " or ""
3. null or undefined
either the check should have been ``` typeof this.stringStack.peek() == undefined ``` or you could have used stack.count?

Reply:
This will work fine as here it is checked as string and “0” is will not be considered as integer 0.
“ “ is also not considered as empty string.
“” will be treated as false as expected.

A sample console log tested with string with 0 and empty spaces “ “:
poppedElement : ), this.count : 0
poppedElement :  , this.count : 1
poppedElement : ), this.count : 1
poppedElement : 0, this.count : 2
poppedElement : -, this.count : 2
poppedElement : 4, this.count : 2
poppedElement : (, this.count : 2
poppedElement :  , this.count : 1
poppedElement : /, this.count : 1
poppedElement : ), this.count : 1
poppedElement : 8, this.count : 2
poppedElement : +, this.count : 2
poppedElement : 3, this.count : 2
poppedElement : (, this.count : 2
poppedElement : *, this.count : 1
poppedElement : 5, this.count : 1
poppedElement : (, this.count : 1
poppedElement : t, this.count : 0
poppedElement : r, this.count : 0
poppedElement : q, this.count : 0
poppedElement : s, this.count : 0

Reply:
mil toh gaya hi hoga falsy wala problem

Reply:
as it is a string, there was no problem here as " " and "0" are also string object received.